pkg/omni/repository/repository_test.go

```diff
@@ -0,0 +1,141 @@
+	Topic3:      "",
+	Data:        "0x000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc200000000000000000000000089d24a6b4ccb1b6faa2625fe562bdd9a23260359000000000000000000000000000000000000000000000000392d2e2bda9c00000000000000000000000000000000000000000000000000927f41fa0a4a418000000000000000000000000000000000000000000000000000000000005adcfebe",
+}
+
+var _ = Describe("Repository Test", func() {
+	var db *postgres.DB
+	var logRepository repositories.LogRepository
+	var blockRepository repositories.BlockRepository
+	var receiptRepository repositories.ReceiptRepository
+	var dataStore repository.DataStore
+	var err error
+	var info *contract.Contract
+	var blockNumber int64
+	var blockId int64
+	var vulcanizeLogId int64
+	rand.Seed(time.Now().UnixNano())
+
+	BeforeEach(func() {
+		db, err = postgres.NewDB(config.Database{
+			Hostname: "localhost",
+			Name:     "vulcanize_private",
+			Port:     5432,
+		}, core.Node{})
+		Expect(err).NotTo(HaveOccurred())
+
+		receiptRepository = repositories.ReceiptRepository{DB: db}
+		logRepository = repositories.LogRepository{DB: db}
+		blockRepository = *repositories.NewBlockRepository(db)
+
+		blockNumber = rand.Int63()
+		blockId = test_helpers.CreateBlock(blockNumber, blockRepository)
+
+		log := core.Log{}
+		logs := []core.Log{log}
+		receipt := core.Receipt{
+			Logs: logs,
+		}
+		receipts := []core.Receipt{receipt}
+
+		err = receiptRepository.CreateReceiptsAndLogs(blockId, receipts)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = logRepository.Get(&vulcanizeLogId, `SELECT id FROM logs`)
+		Expect(err).ToNot(HaveOccurred())
+
+		p := parser.NewParser("")
+		err = p.Parse(constants.TusdContractAddress)
+		Expect(err).ToNot(HaveOccurred())
+
+		info = &contract.Contract{
+			Name:          "TrueUSD",
+			Address:       constants.TusdContractAddress,
+			Abi:           p.Abi(),
+			ParsedAbi:     p.ParsedAbi(),
+			StartingBlock: 5197514,
+			Events:        p.GetEvents(),
+			Methods:       p.GetMethods(),
+			Addresses:     map[string]bool{},
+		}
+
+		event := info.Events["Transfer"]
+		err = info.GenerateFilters([]string{"Transfer"})
+		Expect(err).ToNot(HaveOccurred())
+		c := converter.NewConverter(*info)
+		mockEvent.LogID = vulcanizeLogId
+		err = c.Convert(mockEvent, event)
+		Expect(err).ToNot(HaveOccurred())
+
+		dataStore = repository.NewDataStore(db)
+	})
+
+	AfterEach(func() {
+		db.Query(`DELETE FROM blocks`)
+		db.Query(`DELETE FROM logs`)
+		db.Query(`DELETE FROM transactions`)
+		db.Query(`DELETE FROM receipts`)
+		db.Query(`DROP SCHEMA IF EXISTS trueusd CASCADE`)
+	})
+
+	It("Persist contract info in custom tables", func() {
+		err = dataStore.PersistEvents(info)
+		Expect(err).ToNot(HaveOccurred())
+
+		b, ok := info.Addresses["0x000000000000000000000000000000000000Af21"]
+		Expect(ok).To(Equal(true))
+		Expect(b).To(Equal(true))
+
+		b, ok = info.Addresses["0x09BbBBE21a5975cAc061D82f7b843bCE061BA391"]
+		Expect(ok).To(Equal(true))
+		Expect(b).To(Equal(true))
```

Comment:
It's not really clear to me what's being tested here. Under what conditions would it fail?

Reply:
Currently it is just testing that no errors are thrown during the persist method and that any addresses in the event log are being cached as the log is committed to the repo.  Checking for an address that wasn't in the mockEvent would fail. This needs to be expanded to actually test that the log data was also committed to the expected table.